app/_components/_sections/SavedWorkoutsList.tsx

```diff
@@ -0,0 +1,196 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { FaArrowLeftLong } from "react-icons/fa6";
+import { CiBookmarkMinus } from "react-icons/ci";
+import { CgNotes } from "react-icons/cg";
+import { HiMiniSquares2X2 } from "react-icons/hi2";
+import { FaSpinner } from "react-icons/fa";
+import DeleteWorkoutModal from "../modals/deleteWorkoutModal";
+import { useTranslation } from "react-i18next";
+import { useDeleteWorkout } from "@/app/_services/userService";
+
+interface Workout {
+  _id: string;
+  image_url: string;
+  name: string;
+  level: string;
+  duration?: number;
+  workout_schedule: any[];
+}
+
+interface SavedWorkoutsModalProps {
+  workouts: Workout[];
+  onClose: () => void;
+  refetchSavedWorkouts: () => void;
+}
+
+const calculateTotalDuration = (workout: Workout) => {
+  let totalDurationSeconds = 0;
+
+  workout.workout_schedule?.forEach((schedule) => {
+    schedule.exercises?.forEach((exercise) => {
+      const sets = exercise.sets || 0;
+      const restSeconds = exercise.rest_seconds || 0;
+      const setDurationSeconds = 120;
+
+      const exerciseDurationSeconds = sets * setDurationSeconds;
+      const totalRestBetweenSets = (sets - 1) * restSeconds;
+
+      totalDurationSeconds += exerciseDurationSeconds + totalRestBetweenSets;
+    });
+  });
+
+  return Math.floor(totalDurationSeconds / 60);
+};
+
+const SavedWorkoutsModal: React.FC<SavedWorkoutsModalProps> = ({ workouts, onClose, refetchSavedWorkouts }) => {
+  const { t } = useTranslation("global");
+  const { mutate: deleteWorkout } = useDeleteWorkout();
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedWorkout, setSelectedWorkout] = useState<Workout | null>(null);
+  const [clickedStates, setClickedStates] = useState<{ [key: string]: boolean }>({});
+  const [workoutList, setWorkoutList] = useState<Workout[]>(workouts);
+  const [loadingWorkoutId, setLoadingWorkoutId] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (workouts) {
+      setWorkoutList(workouts);
+      const initialClickedStates = workouts.reduce((acc, workout) => {
+        acc[workout._id] = true;
+        return acc;
+      }, {} as { [key: string]: boolean });
+      setClickedStates(initialClickedStates);
+    }
+  }, [workouts]);
+
+  const handleBookmarkClick = (workout: Workout) => {
+    const duration = calculateTotalDuration(workout);
+    setSelectedWorkout({ ...workout, duration });
+    setModalOpen(true);
+  };
+
+  const handleConfirmDelete = () => {
+    if (selectedWorkout) {
+      deleteWorkout(
+        { queryParams: { workout_id: selectedWorkout._id } },
+        {
+          onSuccess: () => {
+            setClickedStates((prev) => {
+              const updatedStates = { ...prev };
+              delete updatedStates[selectedWorkout._id];
+              return updatedStates;
+            });
+
+            setWorkoutList((prev) => prev.filter(workout => workout._id !== selectedWorkout._id));
```

Comment:
Each deletion triggers a complete re-render of the list due to state updates. Use React.memo or update only the affected workout without filtering the entire list.
tsx

```
setWorkoutList((prev)` => {
  const index = prev.findIndex(({ _id }) => _id === selectedWorkout._id);
  if (index > -1) {
    const updated = [...prev];
    updated.splice(index, 1);
    return updated;
  }
  return prev;
});
```